dask_planner/src/sql/logical.rs

```diff
@@ -206,14 +208,14 @@ impl PyLogicalPlan {
 
     #[pyo3(name = "getRowType")]
     pub fn row_type(&self) -> PyResult<RelDataType> {
-        let schema = self.original_plan.schema();
-        let rel_fields: Vec<RelDataTypeField> = schema
-            .fields()
-            .iter()
-            .map(|f| RelDataTypeField::from(f, schema.as_ref()))
-            .collect::<Result<Vec<_>>>()
-            .map_err(py_type_err)?;
-        Ok(RelDataType::new(false, rel_fields))
+        let mut all_rel_fields: HashSet<RelDataTypeField> = HashSet::new();
+        for schema in self.original_plan.all_schemas() {
+            for field in schema.fields() {
+                all_rel_fields.insert(RelDataTypeField::from(field, schema.as_ref()).unwrap());
```

Comment:
Previously an `Err` rather than an `Ok` would be propagated to the caller via `PyResult`, this instead `panic!`s which is more unfriendly.